### ISSUE TITLE
kube-prometheus: look into using persistent volume(s) for pusher prom statefulset

### ISSUE THREAD

**Issue by jgehrcke** [conbench/conbench#1034]
In https://github.com/conbench/conbench/issues/993#issuecomment-1494157591 I found that Prometheus state was lost (scraped data). I assumed that the `kube-prometheus` stack would automatically create small/default persistent volumes as long as it finds some storage resources to use. But yeah, it can't really pick decent defaults.

Likewise, to confirm (for vd-2):
```
$ kubectl get pvc -n monitoring
No resources found in monitoring namespace.
```

There are great docs about this:

> By default (if the following 'storage.volumeClaimTemplate' isn't created), prometheus will be created with an EmptyDir for the 'prometheus-k8s-db' volume (for the prom tsdb).

(from https://github.com/prometheus-operator/kube-prometheus/blob/7a3879ba49bfe8df5eec03847fe5bcd2d4094c73/examples/prometheus-pvc.jsonnet#L24)

```
$ kubectl get storageclass
NAME            PROVISIONER             RECLAIMPOLICY   VOLUMEBINDINGMODE      ALLOWVOLUMEEXPANSION   AGE
gp2 (default)   kubernetes.io/aws-ebs   Delete          WaitForFirstConsumer   false                  176d
```

Also interesting:

```
                // The following 'selector' is only needed if you're using manual storage provisioning (https://github.com/coreos/prometheus-operator/blob/master/Documentation/user-guides/storage.md#manual-storage-provisioning).
                // And note that this is not supported/allowed by AWS - uncommenting the following 'selector' line (when deploying kube-prometheus to a K8s cluster in AWS) will cause the pvc to be stuck in the Pending status and have the following error:
                //  * 'Failed to provision volume with StorageClass "ssd": claim.Spec.Selector is not supported for dynamic provisioning on AWS'
                // selector: { matchLabels: {} },
```

- https://github.com/prometheus-operator/kube-prometheus/blob/7a3879ba49bfe8df5eec03847fe5bcd2d4094c73/examples/prometheus-pvc.jsonnet
- https://github.com/prometheus-operator/prometheus-operator/issues/2666#issuecomment-510465282
- https://github.com/prometheus-operator/kube-prometheus/issues/883#issuecomment-764628321

**Comment by jgehrcke** [comment]
Clarification: "pusher prom statefulset" is may way of describing the k8s StatefulSet managing N Prometheus instances whose primary job it is to scrape&push, in a best-effort buffering fashion. This StatefulSet is managed by Prometheus Operator.

**Comment by jgehrcke** [comment]
For now resolved by  #1037.

**Comment by jgehrcke** [comment]
Confirmed that after merging #1037 the conbench-staging-deploy/builds/364#01874c08-e800-4c6c-9e37-615655cf6e77 ran fine, updating all k8s objects related to kube-prometheus but this time retaining timeseries database state:

![image](https://user-images.githubusercontent.com/265630/229781632-cb1ce1df-d612-4ca4-b22f-740f4f9c6ac2.png)

This is pretty slick.
